elixir/apps/domain/lib/domain/billing/stripe/api_client.ex

```diff
@@ -98,7 +99,69 @@ defmodule Domain.Billing.Stripe.APIClient do
         :www_form
       )
 
-    request(api_token, :post, "subscriptions", body)
+    request_with_retry(api_token, :post, "subscriptions", body)
+  end
+
+  def request_with_retry(api_token, method, path, body) do
+    max_retries = fetch_retry_config(:max_retries, 3)
+    base_delay = fetch_retry_config(:base_delay_ms, 1000)
+    max_delay = fetch_retry_config(:max_delay_ms, 10_000)
+
+    do_request_with_retry(api_token, method, path, body, 0, max_retries, base_delay, max_delay)
+  end
+
+  defp do_request_with_retry(
+         api_token,
+         method,
+         path,
+         body,
+         attempt,
+         max_retries,
+         base_delay,
+         max_delay
+       ) do
+    case request(api_token, method, path, body) do
+      {:ok, response} ->
+        {:ok, response}
+
+      {:error, {429, response}} when attempt < max_retries ->
+        delay = calculate_delay(attempt, base_delay, max_delay)
+
+        Logger.warning(
+          "Rate limited by Stripe API (429), retrying request.",
+          %{
```

Comment:
Just double checking, maps work here too? I know we typically pass this as a keyword list.